src/Ast.ml

```diff
@@ -1920,6 +1922,13 @@ end = struct
             else exposed_right_exp Non_apply exp
         | _ -> exposed_right_exp Non_apply exp )
     in
+    let rec ifthenelse pexp_desc =
+      match pexp_desc with
+      | Pexp_extension (_, PStr [{pstr_desc= Pstr_eval (e, _)}]) ->
+          ifthenelse e.pexp_desc
```

Comment:
this should be probably use `Source.extension_using_sugar`

Reply:
I don't know how this function is supposed to interact with this match, do we only have to consider cases when `extension_using_sugar` stands? or only the opposite? Am I missing something?